### PR TITLE
fix: authenticate LDAP via actual DN instead of normalized DN 

### LINKED ISSUE #19757: Error auth encoding in LDAP
## Expected Behavior

It is expected that it will be possible to use Cyrillic characters in DN.

## Current Behavior

Unable to login via LDAP with account following DN
sAMAccountName: i.ivanov
CN=Иванов Иван,OU=Users,DC=home,DC=local (Cyrillic symbols)

Upon request to
http://minio.local:9000/?Action=AssumeRoleWithLDAPIdentity&LDAPUsername=i.ivanov&LDAPPassword=*****&Version=2011-06-15

We receive the following error:
```
<?xml version="1.0" encoding="UTF-8"?>
<ErrorResponse xmlns="https://sts.amazonaws.com/doc/2011-06-15/">
    <Error>
        <Type></Type>
        <Code>InvalidParameterValue</Code>
        <Message>LDAP server error: LDAP auth failed for DN cn=**\d0\98\d0\b2\d0\b0\d0\bd\d0\be\d0\b2 \d0\98\d0\b2\d0\b0\d0\bd**,ou=Users,dc=home,dc=local: LDAP Result Code 49 &#34;Invalid Credentials&#34;: 80090308: LdapErr: DSID-0C090439, comment: AcceptSecurityContext error, data 52e, v4563 </Message>
    </Error>
    <RequestId>17D0040BD4CB1718</RequestId>
</ErrorResponse>
```

where \d0\98\d0\b2\d0\b0\d0\bd\d0\be\d0\b2 \d0\98\d0\b2\d0\b0\d0\bd - wrong encoding.

When using CN=test,OU=Users,DC=home,DC=local

```
<AssumeRoleWithLDAPIdentityResponse xmlns="https://sts.amazonaws.com/doc/2011-06-15/">
    <AssumeRoleWithLDAPIdentityResult>
        <Credentials>
            <AccessKeyId>...</AccessKeyId>
            <SecretAccessKey>...</SecretAccessKey>
            <SessionToken>...</SessionToken>
            <Expiration>...</Expiration>
        </Credentials>
    </AssumeRoleWithLDAPIdentityResult>
    <ResponseMetadata>
        <RequestId>..</RequestId>
    </ResponseMetadata>
</AssumeRoleWithLDAPIdentityResponse>
```

## Your Environment
running in docker signle node.

minio version RELEASE.2024-05-10T01-41-38Z (commit-id=b5984027386ec1e55c504d27f42ef40a189cdb55)
Runtime: go1.22.3 linux/amd64
License: GNU AGPLv3 - https://www.gnu.org/licenses/agpl-3.0.html
Copyright: 2015-2024 MinIO, Inc.

Linux 6207cee3b5da 5.15.0-107-generic #117-Ubuntu SMP Fri Apr 26 12:26:49 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

upd 17.05.24: running as service ubuntu, behavior is the same.

### PR DESCRIPTION
**Pull request by harshavardhana**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
fix: authenticate LDAP via actual DN instead of normalized DN 

## Motivation and Context
normalized DN is only for internal representation, not for
external communication, any communication to LDAP must be
based on actual user DN. LDAP servers do not understand
normalized DN.
    
fixes #19757

## How to test this PR?
As per #19757

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by harshavardhana** [comment]
This PR depends on https://github.com/minio/minio/pull/19758 to be merged first

**Comment by donatello** [comment]
It looks like a difference between AD vs OpenLDAP - specifically AD is not accepting DN's in the login with unicode characters escaped as hex values like `\d0`. OpenLDAP does accept this.

**Comment by harshavardhana** [comment]
> It looks like a difference between AD vs OpenLDAP - specifically AD is not accepting DN's in the login with unicode characters escaped as hex values like `\d0`. OpenLDAP does accept this.

yeah but we should use the simplified representation.

File: internal/config/identity/ldap/ldap.go [hunk 1]
```diff
@@ -313,7 +313,11 @@ func (l *Config) GetNonEligibleUserDistNames(userDistNames []string) ([]string,
 		if len(searchResult.Entries) == 0 {
 			// DN was not found - this means this user account is
 			// expired.
-			nonExistentUsers = append(nonExistentUsers, dn)
+			ndn, err := ldap.ParseDN(dn)
```
Comment (taran-p) [inline internal/config/identity/ldap/ldap.go:320]: I think this change needs to go in the above `if err != nil` block as well

Comment (harshavardhana) [reply internal/config/identity/ldap/ldap.go:320]: yes good catch.

**Comment by taran-p** [comment]
We should probably also add the ldapActualUser claim to AddServiceAccount and AddServiceAccountLDAP as well to avoid them getting cleared in the periodic refresh

**Comment by harshavardhana** [comment]
> We should probably also add the ldapActualUser claim to AddServiceAccount and AddServiceAccountLDAP as well to avoid them getting cleared in the periodic refresh

why would they get cleared?

**Comment by harshavardhana** [comment]
> We should probably also add the ldapActualUser claim to AddServiceAccount and AddServiceAccountLDAP as well to avoid them getting cleared in the periodic refresh

I think I understood what you meant, I think I updated the relevant sections properly.

**Comment by harshavardhana** [comment]
@donatello @taran-p PTAL

**Review by taran-p — APPROVED** [review]
LGTM

**Review by donatello — APPROVED** [review]

**Comment by Aliroe** [comment]
中文：我试了一下这个版本从minio version RELEASE.2024-03-26T22-10-45Z升级到minio-05-27 version RELEASE.2024-05-27T19-17-46Z
依然不可用
English：I tried this version upgrade from minio version RELEASE. 2024-03-26 T22-10-45Z to minio-05-27 version RELEASE. 2024-05-27 T19-17-46Z
Still unavailable
image

**Comment by harshavardhana** [comment]
> English：I tried this version upgrade from minio version RELEASE. 2024-03-26 T22-10-45Z to minio-05-27 version RELEASE. 2024-05-27 T19-17-46Z

you cannot downgrade and upgrade that is not supported, please also don't comment on PRs.

**Pull request merged**